src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -90,8 +90,8 @@ internal static class MethodGenerator
 
             var explicitInterfaceSpecifier = GenerateExplicitInterfaceSpecifier(method.ExplicitInterfaceImplementations);
 
-            var returnType = method.ReturnsByRef
-                ? method.ReturnType.GenerateRefTypeSyntax()
+            var returnType = method.ReturnsByRef ? method.ReturnType.GenerateRefTypeSyntax()
+                : method.ReturnsByRefReadonly ? method.ReturnType.GenerateRefReadOnlyTypeSyntax()
                 : method.ReturnType.GenerateTypeSyntax();
```

Comment:
formatting.  also, try to share code with teh other snippet that is identical to this.